### ISSUE TITLE
Different implementation for ResNet

### ISSUE THREAD

**Issue by JAEarly** [okrasolar/inception-time-pytorch#4]
I've noticed some differences to the original ResNet code at https://github.com/cauchyturing/UCR_Time_Series_Classification_Deep_Learning_Baseline/blob/master/ResNet.py.

You're missing at BatchNorm right at the start of the network, and also the residual should be added before applying ReLU in the final layer of each ResNet block, not after applying ReLU. 

In my experiments, adding the above two changes seems to generate results that better match the original results.

**Comment by JAEarly** [comment]
Just to add a bit more to this, I don't think the BatchNorm right at the start is necessary if the data is z-normalised, but adding the residual before applying ReLU definitely improves performance.

**Comment by itakatz** [comment]
Thanks, I also wonder about this line:
https://github.com/okrasolar/pytorch-timeseries/blob/88f65ad7d2bfdf5bea5c5e3aa1eab90c46b1856b/src/models/resnet_baseline.py#L71
Don't we want to add the residual (just `x` in this case) in any case, also when the size matches?
